### PR TITLE
Remove include_cancel in taxes_by_invoice report

### PR DESCRIPTION
**Pull request by raimonesteve**
#157246

### CODE REVIEW

File: taxes_by_invoice/taxes_by_invoice.py [hunk 1]
```diff
@@ -334,6 +332,16 @@ def prepare(cls, data):
 
             for tax in taxes:
                 records.setdefault(tax.tax, []).append(DualRecord(tax))
+
+                # If the invoice is cancelled, do not add its values to the
+                # totals
+                tax.invoice.state == 'cancelled' and tax.invoice.cancel_move
```
Comment (albertca) [inline taxes_by_invoice/taxes_by_invoice.py]: This line should be removed as it does nothing

File: taxes_by_invoice/templates/detail_lines_macro.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {% set nc = namespace(before_invoice_id=None) %}
 {%for l in record %}
     {% if nc.before_invoice_id != l.invoice.raw.id %}
-        <tr>
+        <tr class="{% if l.invoice.raw.state == 'cancelled' %}grey{% endif %}">
```
Comment (albertca) [inline taxes_by_invoice/templates/detail_lines_macro.html:4]: I think we should explain in the report header that cancelled invoices are shown in grey and that those with a cancel move are not added to the total.

Comment (raimonesteve) [reply taxes_by_invoice/templates/detail_lines_macro.html:4]: f2f4a6b40da0d3abeaeafeaccc89f056c1295abb

Add checklist the locales (babel):

Las facturas canceladas se muestran en gris y aquellas con movimiento de cancelación no se suman al total.
Les factures cancel·lades es mostren en gris i les que tenen un moviment de cancel·lació no s'afegeixen al total.

File: taxes_by_invoice/taxes_by_invoice.py (hunk 1 above)
Comment (raimonesteve) [reply taxes_by_invoice/taxes_by_invoice.py]: ups!

f2f4a6b40da0d3abeaeafeaccc89f056c1295abb

File: taxes_by_invoice/taxes_by_invoice.py [hunk 3]
```diff
@@ -302,7 +297,10 @@ def prepare(cls, data):
 
                 # If the invoice is cancelled, do not add its values to the
                 # totals
-                if data['include_cancel'] and tax.invoice.state == 'cancelled':
+                if (tax.invoice.state == 'cancelled'
+                        and tax.invoice.cancel_move
+                        and tax.invoice.cancel_move.origin
+                        and not isinstance(tax.invoice.cancel_move.origin, Invoice)):
```
Comment (bernatnan) [inline taxes_by_invoice/taxes_by_invoice.py]: But if the invoice doesn't have cancel_move or the cancel_move doesn't have origin, in those 2 cases the taxes haven't to be add to the total either.

File: taxes_by_invoice/taxes_by_invoice.py [hunk 4]
```diff
@@ -334,6 +332,15 @@ def prepare(cls, data):
...
             for tax in taxes:
                 records.setdefault(tax.tax, []).append(DualRecord(tax))
+
+                # If the invoice is cancelled, do not add its values to the
+                # totals
+                if (tax.invoice.state == 'cancelled'
+                        and tax.invoice.cancel_move
+                        and tax.invoice.cancel_move.origin
+                        and not isinstance(tax.invoice.cancel_move.origin, Invoice)):
+                    continue
```
Comment (bernatnan) [inline taxes_by_invoice/taxes_by_invoice.py:346]: Idem the above comment.

File: taxes_by_invoice/templates/header_macro.html [hunk 5]
```diff
@@ -39,8 +39,10 @@ <h1 style="font-size:20px">{% if data['parameters']['jump_page'] %}{{ _('Taxes B
             </tr>
         {% endif %}
         <tr>
-            <td>{% if data['parameters']['parties'] %}{{ _('Parties: ') }}{{ data['parameters']['parties']}}{% else %}{{ _('All Parties') }}{% endif %}</td>
-            <td style="text-align: right;">{% if data['parameters']['include_cancel'] %}{{ _('This document includes cancelled invoices marked with a *') }}{% else %}{{ _('This document not includes cancelled invoices') }}{% endif %}</td>
+            <td colspan="2">{% if data['parameters']['parties'] %}{{ _('Parties: ') }}{{ data['parameters']['parties']}}{% else %}{{ _('All Parties') }}{% endif %}</td>
+        </tr>
+        <tr>
+            <td colspan="2"><small>{{ _('Cancelled invoices are shown in grey and that those with a cancel move are not added to the total.') }}</small></td>
```
Comment (bernatnan) [inline taxes_by_invoice/templates/header_macro.html]: The correct text will be:

Cancelled invoices are shown in gray and those without a cancel move or a cancel move not related to an invoice are not added to the total.

File: taxes_by_invoice/taxes_by_invoice.py (hunk 4 above)
Comment (raimonesteve) [reply taxes_by_invoice/taxes_by_invoice.py:346]: 9ea6f09c51f08d48a658527b2809a1ec12494873

File: taxes_by_invoice/templates/header_macro.html (hunk 5 above)
Comment (raimonesteve) [reply taxes_by_invoice/templates/header_macro.html]: 9ea6f09c51f08d48a658527b2809a1ec12494873

File: taxes_by_invoice/taxes_by_invoice.py (hunk 3 above)
Comment (raimonesteve) [reply taxes_by_invoice/taxes_by_invoice.py]: 9ea6f09c51f08d48a658527b2809a1ec12494873

**Review by bernatnan — APPROVED** [review]
LGTM

**Review by bernatnan — APPROVED** [review]
LGTM

**Pull request merged**